dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/processing/CommonQueryRequestMapper.java

```diff
@@ -97,6 +103,22 @@ public CommonParams map( CommonQueryRequest request )
         List<OrganisationUnit> userOrgUnits = dataQueryService.getUserOrgUnits( null, request.getUserOrgUnit() );
         List<Program> programs = getPrograms( request );
 
+        Map<String, String> dimensionsByUid = request.getDimension().stream()
+            .collect( Collectors.toMap( identity(), identity() ) );
+
+        programs.stream()
+            .flatMap( program -> getProgramAttributes( List.of( program ) )
+                .map( BaseIdentifiableObject::getUid )
+                // we need fully qualified dimension identifiers
+                .map( attributeUid -> Pair.of( program, attributeUid ) ) )
+            .forEach( fqDimension -> dimensionsByUid.put(
```

Comment:
Suggestion:
Maybe rename "fqDimension" to "fullDimension" or "fullyQualifiedDimension".
Initially, I could not understand what "fq" stood for.